sensor/common/deploymentenhancer/component_test.go

```diff
@@ -0,0 +1,120 @@
+package deploymentenhancer
+
+import (
+	"testing"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stackrox/rox/sensor/common/clusterentities"
+	"github.com/stackrox/rox/sensor/common/message"
+	"github.com/stackrox/rox/sensor/common/registry"
+	"github.com/stackrox/rox/sensor/common/store"
+	"github.com/stackrox/rox/sensor/common/store/mocks"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+func TestComponent(t *testing.T) {
+	suite.Run(t, &ComponentTestSuite{})
+}
+
+type ComponentTestSuite struct {
+	suite.Suite
+
+	rbacStore *mocks.MockRBACStore
+	srvStore  *mocks.MockServiceStore
+	depStore  *mocks.MockDeploymentStore
+
+	mockCtrl          *gomock.Controller
+	mockStoreProvider store.Provider
+}
+
+func (s *ComponentTestSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+
+	s.rbacStore = mocks.NewMockRBACStore(s.mockCtrl)
+	s.srvStore = mocks.NewMockServiceStore(s.mockCtrl)
+	s.depStore = mocks.NewMockDeploymentStore(s.mockCtrl)
+
+	s.mockStoreProvider = mockStoreProvider{
+		rbac: s.rbacStore,
+		srv:  s.srvStore,
+		dep:  s.depStore,
+	}
+}
+
+func (s *ComponentTestSuite) TearDownTest() {
+	s.T().Cleanup(s.mockCtrl.Finish)
+}
+
+func (s *ComponentTestSuite) TestExtractAndEnrichDeployments() {
+	s.rbacStore.EXPECT().GetPermissionLevelForDeployment(gomock.Any()).AnyTimes()
+	s.srvStore.EXPECT().GetExposureInfos(gomock.Any(), gomock.Any()).AnyTimes()
+	s.depStore.EXPECT().EnhanceDeploymentReadOnly(gomock.Any(), gomock.Any()).AnyTimes()
+	dQueue := make(chan *central.DeploymentEnhancementRequest, 10)
+	de := DeploymentEnhancer{
+		responsesC:       make(chan *message.ExpiringMessage),
+		deploymentsQueue: dQueue,
+		storeProvider:    s.mockStoreProvider,
+	}
+
+	actual := de.enhanceDeployments(generateDeploymentMsg("1", 4))
+
+	s.Len(actual, 4, "Expected %v deployments, got %v", 4, len(actual))
```

Comment:
I'm confused how is this working, because I though you'd need `Lenf` to use the formatting directives. 

Also:
1. When using numbers, use the `%d` directive instead of `%v`.
2. I think the message is useless because `Len` by default will already print out the actual and expected values.

Reply:
That's a good point. I've checked it, and it definitely works identically with `Len` and `Lenf`. Either way, I have removed the error message